src/app/dataset-view/additional-components/data-component/data-component.html

```diff
@@ -168,13 +168,21 @@ <h2 class="box-title align-items-center pb-3 m-0">Query:</h2>
             <h2 class="box-title align-items-center pb-3 pt-3 m-0">Error:</h2>
             <p class="sql-error-message">{{ sqlErrorMarker }}</p>
         </ng-container>
-        <ng-container *ngIf="!sqlErrorMarker">
-            <h2 class="box-title align-items-center pb-3 pt-3 m-0">Result:</h2>
+        <ng-container *ngIf="!sqlErrorMarker && currentSchema && currentData">
+            <h2 class="box-title align-items-center py-3 m-0">Result:</h2>
             <app-dynamic-table
                 [hasTableHeader]="true"
                 [schemaFields]="currentSchema ? currentSchema.fields : []"
                 [dataRows]="currentData || []"
             ></app-dynamic-table>
         </ng-container>
+        <ng-container *ngIf="!sqlErrorMarker || !currentSchema || !currentData">
+            <mat-divider></mat-divider>
+            <h3 class="box-title text-center pt-4 author">
+                The current dataset is empty or need schema. Need
+                <a class="add-data-link" href="#"> add data</a> or
+                <a class="add-schema-link" href="#"> add schema</a>.
+            </h3>
+        </ng-container>
```

Comment:
I think we should distinguish these 2 states - no schema and no data.
We cannot add data, when there is no schema.
So if schema is not defined yet, we must go through a wizard to define `SetPollingSource` or `SetTransform`.

The situation when we have a schema, but no data yet, is different. We should invite the user to pull some data or, maybe, configure periodic automatic pull.